internal/common/client_options.go

```diff
@@ -63,6 +63,9 @@ func setUserAgent(client *autorest.Client, tfVersion, partnerID string, disableT
 	tfUserAgent := fmt.Sprintf("HashiCorp Terraform/%s (+https://www.terraform.io) Terraform Plugin SDK/%s", tfVersion, meta.SDKVersionString())
 
 	providerUserAgent := fmt.Sprintf("%s terraform-provider-azurerm/%s", tfUserAgent, version.ProviderVersion)
+	if features.ThreePointOhBeta() || features.ThreePointOh() { // TODO remove for 3.0
```

Comment:
3.0 Beta will contain 3.0 final when it goes live, so this can become:

```suggestion
	if features.ThreePointOhBeta() { // TODO: remove in 3.0
```